src/io/ioctl.rs

```diff
@@ -66,5 +69,71 @@ pub fn ioctl_fionbio<Fd: AsFd>(fd: Fd, value: bool) -> io::Result<()> {
 #[inline]
 #[doc(alias = "FIONREAD")]
 pub fn ioctl_fionread<Fd: AsFd>(fd: Fd) -> io::Result<u64> {
-    backend::io::syscalls::ioctl_fionread(fd.as_fd())
+    ioctl::ioctl(fd, Fionread(MaybeUninit::uninit()))
+}
+
+#[cfg(apple)]
+struct Fioclex;
+
+#[cfg(apple)]
+#[allow(unsafe_code)]
+unsafe impl ioctl::Ioctl for Fioclex {
+    type Output = ();
+
+    const IS_MUTATING: bool = false;
+    const OPCODE: ioctl::Opcode = ioctl::Opcode::bad(c::FIOCLEX);
+
+    fn as_ptr(&mut self) -> *mut c::c_void {
+        core::ptr::null_mut()
+    }
+
+    unsafe fn output_from_ptr(
+        _: ioctl::IoctlOutput,
+        _: *mut c::c_void,
+    ) -> io::Result<Self::Output> {
+        Ok(())
+    }
+}
+
+struct Fionbio(c::c_int);
+
+#[allow(unsafe_code)]
+unsafe impl ioctl::Ioctl for Fionbio {
+    type Output = ();
+
+    const IS_MUTATING: bool = false;
+    const OPCODE: ioctl::Opcode = ioctl::Opcode::bad(c::FIONBIO);
+
+    fn as_ptr(&mut self) -> *mut c::c_void {
+        (&mut self.0 as *mut c::c_int).cast()
+    }
+
+    unsafe fn output_from_ptr(
+        _: ioctl::IoctlOutput,
+        _: *mut c::c_void,
+    ) -> io::Result<Self::Output> {
+        Ok(())
+    }
+}
```

Comment:
A lot of ioctls are simple "setters" like this, and they'll all return `()`, all be mutating, they'll all do `(&mut self.0 as *mut T)` in their `as_ptr`, and `Ok(())` in their `output_from_ptr`.

Similarly, a lot of ioctls are "getters" which will also have a lot in common with each other.

"getters" and "setters" are common enough that I wonder if it's possible to define some more concise way to define them so that we don't have to repeat all this code so much.